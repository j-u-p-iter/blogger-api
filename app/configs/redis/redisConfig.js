import Joi from 'joi';


export const createRedisConfig = (environmentObject) => {
  const envVarsSchema = Joi.object().keys({
    REDIS_PORT: Joi.number().required(),
    REDIS_HOST: Joi.string().required(),
  }).unknown(true);

  const {
    error,
    value: envVars,
  } = Joi.validate(environmentObject, envVarsSchema);

  if (error) { throw Error(`Redis config validation error: ${error}`); }

  const { REDIS_PORT, REDIS_HOST } = envVars;
  const redisConfig = { REDIS_PORT, REDIS_HOST };

  return redisConfig;
};
