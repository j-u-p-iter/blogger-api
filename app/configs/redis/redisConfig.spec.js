import { createRedisConfig } from './redisConfig';


describe('createRedisConfig(environmentObject)', () => {
  describe('when REDIS_PORT and REDIS_HOST were defined', () => {
    it('should return correct config', () => {
      const REDIS_PORT = 5000;
      const REDIS_HOST = 'redis-host';
      const environmentObject = {
        REDIS_PORT,
        REDIS_HOST,
      };

      const result = createRedisConfig(environmentObject);

      expect(result).toMatchSnapshot();
    });
  });
});
