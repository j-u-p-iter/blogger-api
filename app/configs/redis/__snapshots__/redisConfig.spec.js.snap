// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createRedisConfig(environmentObject) when REDIS_PORT and REDIS_HOST were defined should return correct config 1`] = `
Object {
  "REDIS_HOST": "redis-host",
  "REDIS_PORT": 5000,
}
`;
